packages/integrations/cloudflare/src/index.ts

```diff
@@ -90,33 +93,86 @@ export default function createIntegration(args?: Options): AstroIntegration {
 					vite.ssr.target = 'webworker';
 				}
 			},
+			'astro:build:ssr': ({ manifest, entryPoints }) => {
+				_entryPoints = entryPoints;
+			},
 			'astro:build:done': async ({ pages, routes, dir }) => {
-				const entryPath = fileURLToPath(new URL(_buildConfig.serverEntry, _buildConfig.server));
 				const entryUrl = new URL(_buildConfig.serverEntry, _config.outDir);
 				const buildPath = fileURLToPath(entryUrl);
 				// A URL for the final build path after renaming
 				const finalBuildUrl = pathToFileURL(buildPath.replace(/\.mjs$/, '.js'));
+				const functionsUrl = new URL('functions/', _config.root);
 
-				await esbuild.build({
-					target: 'es2020',
-					platform: 'browser',
-					conditions: ['workerd', 'worker', 'browser'],
-					entryPoints: [entryPath],
-					outfile: buildPath,
-					allowOverwrite: true,
-					format: 'esm',
-					bundle: true,
-					minify: _config.vite?.build?.minify !== false,
-					banner: {
-						js: SHIM,
-					},
-					logOverride: {
-						'ignored-bare-import': 'silent',
-					},
-				});
+				if (isModeDirectory) {
+					await fs.promises.mkdir(functionsUrl, { recursive: true });
+				}
+
+				if (isModeDirectory && _buildConfig.split) {
+					const entryPaths = [..._entryPoints.values()].map((entry) => fileURLToPath(entry));
+					const outputDir = fileURLToPath(new URL('.astro', _buildConfig.server));
+
+					// NOTE: AFAIK, esbuild keeps the order of the entryPoints array
+					const { outputFiles } = await esbuild.build({
+						target: 'es2020',
+						platform: 'browser',
+						conditions: ['workerd', 'worker', 'browser'],
+						entryPoints: entryPaths,
+						outdir: outputDir,
+						allowOverwrite: true,
+						format: 'esm',
+						bundle: true,
+						minify: _config.vite?.build?.minify !== false,
+						banner: {
+							js: SHIM,
+						},
+						logOverride: {
+							'ignored-bare-import': 'silent',
+						},
+						write: false,
+					});
+
+					for (const [index, outputFile] of outputFiles.entries()) {
+						const fileName = [..._entryPoints.entries()][index][0].component.replace('src/pages/', '').replace('.astro', '.js').replace(/(\[\.\.\.)(\w+)(\])/g, (_match, _p1, p2, _p3) => {
```

Comment:
Tip: you're creating a new array at every loop iteration, you can move `[..._entryPoints.entries()]` outside of the loop.